storage/pipeline/states_sealing.go

```diff
@@ -749,109 +713,71 @@ func (m *Sealing) handleSubmitCommit(ctx statemachine.Context, sector SectorInfo
 // processPieces returns either:
 // - a list of piece activation manifests
 // - a list of deal IDs, if all non-filler pieces are deal-id pieces
-func (m *Sealing) processPieces(ctx context.Context, sector SectorInfo, forceDDO bool) ([]miner.PieceActivationManifest, []abi.DealID, error) {
+func (m *Sealing) processPieces(ctx context.Context, sector SectorInfo) ([]miner.PieceActivationManifest, bool, error) {
 	pams := make([]miner.PieceActivationManifest, 0, len(sector.Pieces))
-	dealIDs := make([]abi.DealID, 0, len(sector.Pieces))
-	hasDDO := forceDDO
-
-	if !forceDDO {
-		// if not forcing DDO, check if we have any DDO pieces
-		for _, piece := range sector.Pieces {
-			piece := piece
-
-			// first figure out if this is a ddo sector
-			err := piece.handleDealInfo(handleDealInfoParams{
-				FillerHandler: func(info UniversalPieceInfo) error {
-					// Fillers are implicit (todo review: Are they??)
-					return nil
-				},
-				BuiltinMarketHandler: func(info UniversalPieceInfo) error {
-					return nil
-				},
-				DDOHandler: func(info UniversalPieceInfo) error {
-					hasDDO = true
-					return nil
-				},
-			})
-			if err != nil {
-				return nil, nil, xerrors.Errorf("handleDealInfo: %w", err)
-			}
-		}
-	}
+	var preCommit bool
+
 	for _, piece := range sector.Pieces {
 		piece := piece
+		if piece.HasDealInfo() {
+			info := piece.DealInfo()
+			// If we have a dealID then covert to PAM
+			if info.Impl().DealID > 0 {
```

Comment:
So the two cases, if we have DealInfo, is that `DealID>0` which means we have explicit deal info and need to build a PieceActivationManifest from it ourselves, or the `DealID==0` case which means we've been supplied a PieceActivationManifest and should just use that and ignore the other deal properties?

Reply:
Piece are not only deal pieces here. They can be fillers for partially filled sectors. Thus the check if we have deal info or not.